examples/mnist.py

```diff
@@ -100,29 +110,35 @@ def create_iter_functions(dataset, output_layer,
                           X_tensor_type=T.matrix,
                           batch_size=BATCH_SIZE,
                           learning_rate=LEARNING_RATE, momentum=MOMENTUM):
+    """Create functions for training, validation and testing to iterate one
+       epoch.
+    """
     batch_index = T.iscalar('batch_index')
     X_batch = X_tensor_type('x')
     y_batch = T.ivector('y')
-    batch_slice = slice(
-        batch_index * batch_size, (batch_index + 1) * batch_size)
+    batch_slice = slice(batch_index * batch_size,
+                        (batch_index + 1) * batch_size)
 
     objective = lasagne.objectives.Objective(output_layer,
         loss_function=lasagne.objectives.categorical_crossentropy)
 
-    loss_train = objective.get_loss(X_batch, target=y_batch)
-    loss_eval = objective.get_loss(X_batch, target=y_batch,
-                                   deterministic=True)
+    loss = {'train': objective.get_loss(X_batch, target=y_batch),
+            'eval': objective.get_loss(X_batch,
+                                       target=y_batch,
+                                       deterministic=True)
+           }
```

Comment:
Personal preference :-)  Is there a disadvantage?

I thought I could generate the iter-functions in a more compact way like

```python
iters = {}
for modes in ['train', 'valid']:
    iters[mode] = theano.function(
        [batch_index], loss[mode],
        updates=updates,
        givens={
            X_batch: dataset['X_train'][batch_slice],
            y_batch: dataset['y_train'][batch_slice],
        },
    )
```

as the code is very similar, but then I noticed that it is `loss[mode]` for train and `[loss[mode], accuracy]` for the others.

I like it this way, because it groups two different types of loss functions in one object, now two as it was before.

Should I undo it?